bundles/persistence/org.openhab.persistence.influxdb/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java

```diff
@@ -465,46 +484,87 @@ private String stateToString(State state) {
    * 
    * @param value to be converted to a {@link State}
    * @param itemName name of the {@link Item} to get the {@link State} for
-   * @return
+   * @return the state of the item represented by the itemName parameter, 
+   * 			else the string value of the Object parameter
    */
   private State objectToState(Object value, String itemName) {
     String valueStr = String.valueOf(value);
     if (itemRegistry != null) {
       try {
         Item item = itemRegistry.getItem(itemName);
-        if (item instanceof SwitchItem && !(item instanceof DimmerItem)) {
-          return string2DigitalValue(valueStr).equals(DIGITAL_VALUE_OFF)
-              ? OnOffType.OFF
-              : OnOffType.ON;
+        
+        if (item instanceof ColorItem || item instanceof DimmerItem) {
+        	return item.getState();
+        } else if (item instanceof SwitchItem) {
+        	return convertToSwitchOutput(valueStr).equals(DIGITAL_VALUE_OFF)
```

Comment:
Replacing string2DigitalValue with convertToSwitchOutput and convertToContactOutput is a bad idea because OpenClosedType is stored as 0 or 1. Furthermore I prefer the overhead of the Bigdecimal approach over the exception handling of the Integer.parseInt().